pkg/config/config.go

```diff
@@ -48,12 +54,117 @@ func FromData(yamlData string) (*GithubConfig, error) {
+	}
+
+	// Step 2: delete all secrets that are no more existing
+	found := false
+	for _, existentSec := range secs.Secrets {
+		for _, newSec := range goa.Secrets {
+			if newSec == existentSec.Name {
+				found = true
+				break
+			}
+		}
+		if !found {
+			logrus.Infof("deleting secret '%s' for org '%s'...", existentSec.Name, orgName)
+			_, err = service.DeleteOrgSecret(ctx, orgName, existentSec.Name)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	keyBytes, err := base64.StdEncoding.DecodeString(pKey.GetKey())
+	if err != nil {
+		return err
+	}
+
+	// Step 3: add or update all conf-listed secrets
+	for _, secName := range goa.Secrets {
+		logrus.Infof("adding/updating secret '%s' in org '%s'...", secName, orgName)
+		secValue, err := provider.GetSecret(secName)
+		if err != nil {
+			return err
+		}
+
+		secBytes := sodium.Bytes(secValue)
+		encSecBytes := secBytes.SealedBox(sodium.BoxPublicKey{Bytes: keyBytes})
+		encSecBytesB64 := base64.StdEncoding.EncodeToString(([]byte)(encSecBytes))
+		_, err = service.CreateOrUpdateOrgSecret(ctx, orgName, &github.EncryptedSecret{
+			Name:           secName,
+			KeyID:          pKey.GetKeyID(),
+			EncryptedValue: encSecBytesB64,
+		})
+		if err != nil {
+			return err
+		}
+	}
+	logrus.Infof("secrets synced for org %s\n", orgName)
+	return nil
+}
+
+func (goa *GitHubOrgActions) syncVariables(ctx context.Context,
+	service poiana.ActionsVarsService,
+	orgName string) error {
+	// Step 1: load repo variables
+	logrus.Infof("listing variables for org '%s'...", orgName)
+	vars, _, err := service.ListOrgVariables(ctx, orgName, nil)
+	if err != nil {
+		return err
+	}
+
+	// Step 2: delete all variables that are no more existing
+	for _, existentVar := range vars.Variables {
+		_, ok := goa.Variables[existentVar.Name]
+		if !ok {
+			logrus.Infof("deleting variable '%s' for org '%s'...", existentVar.Name, orgName)
+			_, err = service.DeleteOrgVariable(ctx, orgName, existentVar.Name)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	// Step 3: add or update all conf-listed variables
+	for newVarName, newVarValue := range goa.Variables {
+		logrus.Infof("adding/updating variable '%s' in org '%s'...", newVarName, orgName)
+		resp, err := service.UpdateOrgVariable(ctx, orgName, &github.ActionsVariable{
+			Name:  newVarName,
+			Value: newVarValue,
+		})
+		if err != nil {
+			// Update returns StatusNoContent when successful
+			if resp.StatusCode != http.StatusNoContent {
+				_, err = service.CreateOrgVariable(ctx, orgName, &github.ActionsVariable{
+					Name:  newVarName,
+					Value: newVarValue,
+				})
+			}
+			if err != nil {
+				return err
+			}
+		}
+	}
+	logrus.Infof("variables synced for org %s\n", orgName)
+	return nil
+}
+
+func (gra *GitHubRepoActions) syncSecrets(ctx context.Context,
 	service poiana.ActionsSecretsService,
 	provider poiana.SecretsProvider,
 	pKey *github.PublicKey,
-	orgName, repoName string,
-	secrets []string) error {
+	orgName, repoName string) error {
```

Comment:
We don't need to take `secrets` param anymore, because we have now full access to the receiver.